### PR TITLE
[#247] Syncing Atom DB Python with C++ implementation

### LINKED ISSUE #247: Sync `hyperon-das-atomdb` (Python package) with `hyperon-das-atomdb-cpp` (C++ ram-only and base implementaion)
Atom DB RAM-only was re-implemented in C++, as well as some basic resources like Exceptions, Expression Hasher, AtomDB base/abstract class, among others.
However, a pretty similar Python implementation already existed. And it contains the Redis/Mongo implementation, that won't be re-implemented in C++ as of now.
The ask here is to sync the Python implementation and make it use the C++ implementation module, which was exposed as a Python package with the use of `nanobind`, and eliminate all duplicated resources/features that are also implemented in Python.

### PR DESCRIPTION
**Pull request by angeloprobst**
Closes #247

### CODE REVIEW

File: hyperon_das_atomdb/utils/expression_hasher.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ class StringExpressionHasher:  # TODO(angelo,andre): remove this class? it's not
     """Utility class for generating string representations of expression hashes."""
```
Comment (andre-senna) [inline hyperon_das_atomdb/utils/expression_hasher.py:123]: StringExpressionHasher is meant to be a hasher that generates human-readable unique strings. Something like the way we usually write nodes and links like "<concept: human>" etc. It's supposed to be used in tests, to make debugging easier but I suppose it's not used anymore. Anyway we get used to the regular hashes at the point of recognizing some of the more common ones :-P so I believe this class can be safely removed.

**Review by andre-senna — APPROVED** [review]

File: hyperon_das_atomdb/adapters/redis_mongo_db.py
Comment (eddiebrissow) [inline hyperon_das_atomdb/adapters/redis_mongo_db.py:1]: Replace string with the Enum on line 429 from
```                document["targets"] = self._get_document_keys(document)```
to 
```                document[FieldNames.TARGETS] = self._get_document_keys(document)```

**Review by eddiebrissow — APPROVED** [review]

File: hyperon_das_atomdb/adapters/redis_mongo_db.py
Comment (angeloprobst) [reply hyperon_das_atomdb/adapters/redis_mongo_db.py:1]: Done.

File: hyperon_das_atomdb/utils/expression_hasher.py (hunk 1 above)
Comment (angeloprobst) [reply hyperon_das_atomdb/utils/expression_hasher.py:123]: I noticed it was not being used but was not sure if it could be safely removed. Will do it.

**Pull request merged**